### PR TITLE
Add distributed training overview doc

### PR DESCRIPTION
**Pull request by jacquesqiao**
refs: https://github.com/PaddlePaddle/Paddle/pull/9932
refs: https://docs.google.com/document/d/1-W3HHC0yNYolGuYL7L0Crxs3Us8rZfGjlSe0o5X7lQA/edit#heading=h.td8lf867c0rd

task list: https://github.com/PaddlePaddle/Paddle/issues/9941

### CODE REVIEW

File: doc/fluid/design/dist_train/distributed_training_overview.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+Currently Paddle Fluid use parameter server architecture to support distributed training.
+
+For synchronous and asynchronous training, the differences are mostly in the logic of parameter server. Now we have already support synchronous training.
+
+### Synchronous training
+
+The training process of synchronous training is:
+
+![lookup table training](./src/sync_distributed_training.png)
```
Comment (typhoonzero) [inline doc/fluid/design/dist_train/distributed_training_overview.md]: Better to use a "cross functional flowchart" to represent both procedure and communications.

File: doc/fluid/design/dist_train/distributed_training_overview.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+## distributed training overview doc
```
[the remark this answers is not in the document]
Comment (jacquesqiao) [reply doc/fluid/design/dist_train/distributed_training_overview.md]: Done

File: doc/fluid/design/dist_train/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+Currently Paddle Fluid use parameter server architecture to support distributed training.
+
+For synchronous and asynchronous training, the differences are mostly in the logic of parameter server. Now we have already support synchronous training.
+
+### Synchronous training
+
+The training process of synchronous training is:
+
+![lookup table training](./src/sync_distributed_training.png)
```
Comment (Yancey1989) [inline doc/fluid/design/dist_train/README.md]: `lookup table training ` => synchronous distributed training.

File: doc/fluid/design/dist_train/README.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+In the above process. There are two barriers for all trainers to synchronize with each other. In asynchronous training, these two barriers are not needed. The trainer can just send gradients to pserver and then get parameters back.
+
+The training process of asynchronous training can be:
+
+![lookup table training](./src/async_distributed_training.png)
+
+1. Pserver:
+	1. Each parameter has a queue to receive its gradient from trainers.
+	1. Each parameter has a thread to read data from the queue and run optimize block, using the gradient to optimize the parameter.
+	1. Use a independente thread to handle RPC call `GetVariable` for trainers to get parameters back.(Maybe here we should use a thread pool to speed up the parameter fetch.)
```
Comment (Yancey1989) [inline doc/fluid/design/dist_train/README.md]: Use a independente thread => Using an independent thread

Comment (Yancey1989) [reply doc/fluid/design/dist_train/README.md]: speed up the parameter fetch => speed up fetching the parameters.

File: doc/fluid/design/dist_train/README.md (hunk 3 above)
Comment (jacquesqiao) [reply doc/fluid/design/dist_train/README.md]: done

File: doc/fluid/design/dist_train/README.md (hunk 4 above)
Comment (jacquesqiao) [reply doc/fluid/design/dist_train/README.md]: done

File: doc/fluid/design/dist_train/README.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+Currently Paddle Fluid use parameter server architecture to support distributed training.
+
+For synchronous and asynchronous training, the differences are mostly in the logic of parameter server. Now we have already support synchronous training.
+
+### Synchronous training
+
+The training process of synchronous training is:
+
+![synchronous distributed training](./src/sync_distributed_training.png)
+
```
[the remark this answers is not in the document]
Comment (jacquesqiao) [reply doc/fluid/design/dist_train/README.md]: fixed, thanks! @tonyyang-svail

**Pull request merged**